### PR TITLE
fix: make "EntityType" type compatible with objects

### PR DESCRIPTION
**Pull request by imyasha**
Fixes https://github.com/mswjs/data/issues/250

### CODE REVIEW

**Review by kettanaito — CHANGES_REQUESTED** [review]
Hey, @imyasha 👋 Thanks for opening this.

Looks like the build is failing. I suspect that the `AnyObject` type is not publicly exported by the library (it's internal) and thus certain types cannot compile? 

If you have a minute, could you take a look at what's going on? Thanks.

**Comment by imyasha** [comment]
Hey, @kettanaito 
I was trying to export `AnyObject` but it doesn't solve the build error.
I think it's related with other thing.

Please take a look at it if you can.
Thanks

File: src/query/queryTypes.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ export type GetQueryFor<ValueType extends any> = ValueType extends string
   ? BooleanQuery
   : ValueType extends Date
   ? DateQuery
-  : ValueType extends Array<infer ItemType>
+  : ValueType extends Array<infer ItemType extends AnyObject>
```
Comment (cre8ive-gds) [inline src/query/queryTypes.ts:98]:
```suggestion
  : ValueType extends Array<infer ItemType>
```

**Review by kettanaito — APPROVED** [review]
With @cre8ive-gds additions this looks good now! Thank you!

**Comment by kettanaito** [comment]
## Released: v0.11.2 🎉

This has been released in v0.11.2!

- 📄 [**Release notes**](https://github.com/mswjs/data/releases/tag/v0.11.2)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/data/v/0.11.2)

Make sure to always update to the latest version (`npm i @mswjs/data@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**